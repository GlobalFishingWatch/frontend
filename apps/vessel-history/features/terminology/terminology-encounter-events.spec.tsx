import { Trans, useTranslation } from 'react-i18next'
import { render } from '@testing-library/react'

import TerminologyEncounterEvents from './terminology-encounter-events'

jest.mock('react-i18next')

describe('TerminologyEncounterEvents', () => {
  const mockUseTranslation: jest.Mock = useTranslation as jest.Mock
  const mockTrans: jest.Mock = Trans as jest.Mock
  beforeEach(() => {
    mockUseTranslation.mockReturnValue({
      t: (_ns: any, fallback: any) => fallback,
    })
    mockTrans.mockImplementation(({ i18nKey, children }) => children)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should render successfully', () => {
    const { baseElement } = render(<TerminologyEncounterEvents />)
    expect(baseElement).toMatchSnapshot()
  })
})
