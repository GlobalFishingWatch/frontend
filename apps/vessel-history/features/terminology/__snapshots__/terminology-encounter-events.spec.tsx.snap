// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerminologyEncounterEvents should render successfully 1`] = `
<body>
  <div>
    <p>
      Encounters may indicate potential transshipment activity between two vessels that both appear on AIS. Encounters are estimated using AIS data, including distance between the two vessels, vessel speeds, and duration in a given area. 
      <br />
      An encounter is defined as when two vessels, a carrier vessel and encountered fishing vessel, are detected on AIS data as within 500 meters for at least two hours and traveling at a median speed &lt;2 knots, while at least 10 km from a coastal anchorage.
    </p>
  </div>
</body>
`;
